code/datums/jobs/job/shipside.dm

```diff
@@ -668,6 +668,8 @@ Use your office fax machine to communicate with corporate headquarters or to acq
 		I.update_label()
 	H.name = H.get_visible_name()
 
+	H.equip_to_slot(H.wear_ear, SLOT_EARS) // re-equip headset to fix issues with updated names.
```

Comment:
Oh my I've created a monster. A lot of this proc should preferrably be done via rank assignment rather than equip. Equip should only handle clothing and well, equipment. Are you for refactoring it? If not I'll incorporate it in #2104

Reply:
Not really for now im in bugfix mentality.

Reply:
Well I'll just incorporate the bugfix since I'm refactoring this anyway. Will credit ya.